### PR TITLE
CSD-143: Added support for completion to be called with tagEvent.

### PR DESCRIPTION
**Pull request by LamarDataSherpas**
https://thedatasherpas.atlassian.net/browse/CSD-143
https://thedatasherpas.atlassian.net/browse/CSD-243
CSD-243: Correct RequiredProperties will be sent with install event.

### CODE REVIEW

**Review by QasimAbbas — APPROVED** [review]

File: SiftSdkTests/PlanTesting.swift [hunk 1]
```diff
@@ -83,17 +83,23 @@ class PlanTesting: XCTestCase {
...
       XCTFail(error.localizedDescription)
     }
     
-    let expectation = self.expectation(description: "give it some time to finish, I'll fix this later too busy right now to add support for this")
-    DispatchQueue.global().async {
-      sleep(5)
-      expectation.fulfill()
-    }
-    waitForExpectations(timeout: 15.0)
+    waitForExpectations(timeout: 10.0)
```
Comment (QasimAbbas) [inline SiftSdkTests/PlanTesting.swift:103]: Is this a bit long to wait for the expectation?

**Review by QasimAbbas — CHANGES_REQUESTED** [review]

File: SiftSdkTests/PlanTesting.swift (hunk 1 above)
Comment (LamarDataSherpas) [reply SiftSdkTests/PlanTesting.swift]: changed wait to 5 seconds

File: SiftSdkTests/PlanTesting.swift [hunk 2]
```diff
@@ -83,17 +83,23 @@ class PlanTesting: XCTestCase {
...
     } catch {
       XCTFail(error.localizedDescription)
     }
     
-    let expectation = self.expectation(description: "give it some time to finish, I'll fix this later too busy right now to add support for this")
-    DispatchQueue.global().async {
-      sleep(5)
-      expectation.fulfill()
-    }
-    waitForExpectations(timeout: 15.0)
```
Comment (QasimAbbas) [inline SiftSdkTests/PlanTesting.swift:97]: Looks like this was changed to 15s

File: SiftSdkTests/PlanTesting.swift [hunk 3]
```diff
@@ -83,17 +83,23 @@ class PlanTesting: XCTestCase {
...
+          expectation.fulfill()
+        }
+      }
     } catch {
       XCTFail(error.localizedDescription)
     }
     
-    let expectation = self.expectation(description: "give it some time to finish, I'll fix this later too busy right now to add support for this")
-    DispatchQueue.global().async {
-      sleep(5)
```
Comment (QasimAbbas) [inline SiftSdkTests/PlanTesting.swift:94]: Should expectation be passed with wait?

**Review by QasimAbbas — CHANGES_REQUESTED** [review]

**Review by QasimAbbas — APPROVED** [review]

**Pull request merged**